src/common/util.cpp

```diff
@@ -1063,4 +1064,26 @@ std::string get_nix_version_display_string()
     strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S", &tm);
     return std::string(buffer);
   }
+
+  std::string get_human_readable_bytes(uint64_t bytes)
+  {
+    // Use 1024 for "kilo", 1024*1024 for "mega" and so on instead of the more modern and standard-conforming
+    // 1000, 1000*1000 and so on, to be consistent with other Monero code that also uses base 2 units
+    static const char *SIZES[] = { "B", "kB", "MB", "GB", "TB" };
+    if (bytes < 1024)
+    {
+      return boost::lexical_cast<std::string>(bytes) + " " + SIZES[0];
+    }
```

Comment:
Is this needed ? The loop below should work without this special case (probably changed a bit).

Reply:
Yeah, that special case sure is not elegant, but if I want to format numbers below 1024 in a different way i.e. without 2 decimals I think I need to have an "if" somewhere, and it does not matter much where it is. But maybe I am just lacking a good idea?

Reply:
Ah, OK.

Reply:
I felt like trying it :P 

<pre>
#include <stdio.h>
#include <stdlib.h>


int main(int argc, const char **argv)
{
  double L;
  static const char *units[] = { "B", "kB", "MB", "GB", "TB"};

  if (argc <= 1)
  {
    printf("parameter needed\n");
    return 1;
  }
  int unit_idx = 0;
  L = strtoull(argv[1], NULL, 10);
  int p = 0;
  while (L >= 1024)
  {
    L /= 1024;
    ++unit_idx;
    p = 2;
  }
  printf("%.*f %s\n", p, (float)L, units[unit_idx]);

  return 0;
}

</pre>

Reply:
Hmm, yes, pretty clever how you use the capabilities of `printf` here :)

There seems to be no corresponding trick with `boost::format` however, and I don't think the problem is bad enough to merit a fallback to pure C ...

Reply:
Sure. Also it did not check for overflow in case of petabytes etc :)